extensions/positron-python/src/client/positron/discoverer.ts

```diff
@@ -159,3 +147,41 @@ async function hasFiles(includes: string[]): Promise<boolean> {
     // Exclude node_modules for performance reasons
     return (await vscode.workspace.findFiles(include, '**/node_modules/**', 1)).length > 0;
 }
+
+/**
+ * Recommends a Python interpreter based on workspace configuration and available suggestions.
+ *
+ * The function follows this priority order:
+ * 1. If no workspace and default interpreter is set, use the default interpreter
+ * 2. Use the recommended interpreter from suggestions
+ * 3. Fall back to the active interpreter
+ *
+ * @param serviceContainer - Container providing access to interpreter services
+ * @returns Promise resolving to a recommended Python environment or undefined if none found
+ *
+ */
+export async function recommendInterpreter(
+    serviceContainer: IServiceContainer,
+): Promise<PythonEnvironment | undefined> {
+    const interpreterService = serviceContainer.get<IInterpreterService>(IInterpreterService);
+    const interpreterSelector = serviceContainer.get<IInterpreterSelector>(IInterpreterSelector);
+
+    // Get the recommended interpreter
+    // NOTE: We may need to pass a resource to getSettings to support multi-root workspaces
+    const workspaceUri = vscode.workspace.workspaceFolders?.[0]?.uri;
+    const suggestions = interpreterSelector.getSuggestions(workspaceUri);
+
+    const recommendedInterpreter =
+        // if there there is no workspace and a default interpreter is set, use that
+        (!workspaceUri && getUserDefaultInterpreter()
```

Comment:
this is the only real change here, the rest is just making the discoverer a bit more modular